src/Map/ProspectorOverlayLayer.cs

```diff
@@ -306,19 +403,28 @@ private void OnChatMessage(int groupId, string message, EnumChatType chattype, s
             if (pos == null || groupId != GlobalConstants.InfoLogChatGroup || !_triggerwords.Any(triggerWord => message.StartsWith(triggerWord)))
                 return;
 
-            message = _cleanupRegex.Replace(message, string.Empty);
             var posX = pos.X / _chunksize;
             var posZ = pos.Z / _chunksize;
-            _messages.RemoveAll(m => m.X == posX && m.Z == posZ);
-            _messages.Add(new ProspectInfo(posX, posZ, message));
-            _clientApi.SaveDataFile(Filename, _messages);
+            _prospectInfos.RemoveAll(m => m.X == posX && m.Z == posZ);
+            _prospectInfos.Add(new ProspectInfo(posX, posZ, message));
+            _clientApi.SaveDataFile(Filename, _prospectInfos);
 
             _components.RemoveAll(component =>
             {
                 var castComponent = component as ProspectorOverlayMapComponent;
                 return castComponent?.ChunkX == posX && castComponent.ChunkZ == posZ;
             });
-            var newComponent = new ProspectorOverlayMapComponent(_clientApi, posX, posZ, message, _colorTexture);
+
+            RelativeDensity densityValue;
+            if (_config.HeatMapOre == null)
+                if (_prospectInfos.Last().Values != null && _prospectInfos.Last().Values.Count > 0)
```

Comment:
what exactly does this do ? 
why Last() and not the freshly created ProspectInfo?

Reply:
Yea true 
will change